libraries/psibase/common/include/psibase/Table.hpp

```diff
@@ -271,12 +271,30 @@ namespace psibase
    {
       std::span<const char> data;
    };
+
+   /// A split key is used to specify a subindex on indices with a struct key.
+   ///
+   /// For example, a primr
+   /// Warning: No type checking! It's up to the user to ensure that T + Rest
+   ///   is equivalent to the primary key)
+   template <typename Rest, typename T>
+   struct SplitKey
+   {
+      T value;
+   };
+
```

Comment:
The reason I suggested `SplitKey` is that it's non-intrusive. If we're adding this to the main `Table.hpp` header, a better solution would be to add an overload of `subindex` that takes an explicit type.
`subindex<Rest>(key)`

Reply:
I decided to make this non-intrusive rather than try to figure out how to implement your subindex idea